### PR TITLE
Update DG to include manual testing instructions

### PR DESCRIPTION
**Pull request by soons1**
Add implementations of add, edit and delete

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -213,8 +305,8 @@ The following sequence diagram shows how a Find operation goes through the `Logi
 **Aspect: Tag Parsing in FindCommandParser:** <br>
 
 * **Alternative 1: (Current Choice):** Parsing tags separately from other search criteria.
-  * Pros: Allows users to specify additional search criteria based on tags.
-  * Cons: Requires additional parsing logic and potentially increases complexity in handling multiple search parameters.
+  * Pros: Allows users to specify Deleteitional search criteria based on tags.
+  * Cons: Requires Deleteitional parsing logic and potentially increases complexity in handling multiple search parameters.
```
Comment (CJerrong) [inline docs/DeveloperGuide.md]: This part may have been accidentally overwritten when replacing add with delete.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -753,24 +845,122 @@ testers are expected to do more *exploratory* testing.
...
+    1. Prerequisites: List all persons using the `list` command. Multiple persons in both the befriendees and volunteers list.
+
+    1. Test case: `add n/Michelle r/volunteer p/98909221 e/mich@gmail.com a/Wow street, blk 123, t/smart`<br>
+       Expected: Contact with the above details is added to the volunteer list. Details of the added contact shown in the status message.
+
+    1. Test case: `add n/Giggs p/88221143 e/gig@hotmail.com a/Hello street, block 4, #04-01`<br>
+       Expected: No person is added. Error details indicating "Invalid command format!" shown in the status message. Status bar remains the same.
+
+    1. Other incorrect add commands to try: `add`, `add n/`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.
```
Comment (CJerrong) [inline docs/DeveloperGuide.md]: The part about x may have been accidentally included.

**Review by CJerrong — COMMENTED** [review]
May just have to fix a few small issues and then it'll be good to go

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (soons1) [reply docs/DeveloperGuide.md]: Fixed, thanks!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (soons1) [reply docs/DeveloperGuide.md]: Right, edited!

**Review by CJerrong — APPROVED** [review]
LGTM!

**Pull request merged**